src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -828,8 +828,7 @@ void SolverInterfaceImpl::setMeshTriangle(
 {
   PRECICE_TRACE(meshID, firstEdgeID,
                 secondEdgeID, thirdEdgeID);
-  PRECICE_CHECK(_dimensions == 3, "setMeshTriangle is only possible for 3D cases."
-                                  " Please set the dimension to 3 in the preCICE configuration file.");
+
```

Comment:
@boris-martin allow me a question from my naive point of view: how does setting triangles make sense in case a nearest-projection mapping was configured? Shouldn't this check be conditionally disabled?

Reply:
Right now it doesn't make sense to set triangles in NP mapping indeed. But "3D" is no longer a sufficient condition to disable triangles.
In the long run I think they'll be more common anyways, for instance if we want to allow watch integrals over volume instead of surfaces.

In the case of setMeshTriangle, the connectivity requirement will be reworked to be more specific, so the `context.meshRequirement == mapping::Mapping::MeshRequirement::FULL` will be replaced with a more fine-grained thing. See https://github.com/precice/precice/pull/1203 (which is a draft I'll have to rework)